src/main/java/com/google/devtools/build/lib/rules/cpp/CppConfiguration.java

```diff
@@ -324,6 +325,11 @@ public DynamicMode getDynamicModeFlag() {
     return cppOptions.dynamicMode;
   }
 
+  public boolean isRenameDLL() {
+    return cppOptions.renameDLL && (cppOptions.dynamicMode == DynamicMode.FULLY)
+        && (OS.getCurrent() == OS.WINDOWS);
```

Comment:
I think we shouldn't check `cppOptions.dynamicMode == DynamicMode.FULLY` because even if the dynamic mode is auto, cc_library can still be built dynamically.

And if you want to check the platform we're building for, it's better to use `featureConfiguration.isEnabled(CppRuleClasses.TARGETS_WINDOWS)` if possible. Because in the future, Bazel may runs on Linux but builds for Windows.

Reply:
So I will check `cppOptions.dynamicMode != DynamicMode.OFF`